### DISCUSSION TITLE
Variable inside the console command

### DISCUSSION THREAD

**Discussion by sm1000** [ipxe/ipxe#194] [Q&A] [answered]
The console command in my script works fine:
**console -x 1024 -y 768**

If I try to **set a variable** to hold all options of the **console** command, the execution fails because the variable is recognized as a text value.

Example:
**set space:hex 20 && set space ${space:string}**
**set console-conf -x${space}1024${space}-y${space}768**
**console ${console-conf}**

Error message:
**" 1024 -y 768": invalid integer value**
**Could not boot: Invalid argument (http://ipxe.org/err/1c162102)**

Is there a work-around to pass the console options as variable?

**Comment by mcb30** [comment, marked as answer]
Unfortunately the very simple command parser in iPXE can't support that.  The command line gets tokenized before variable expansion takes place, so `${...}` will always be expanded as (part of) a single argument.  This is what allows your `${space}` trick to work in setting the value of `${console-conf}` in the first place.
